### ISSUE TITLE
Building SIRIUS with CUDA and ELPA fails with: 'elpa_eigenvectors_d' was not declared in this scope

### ISSUE THREAD

**Issue by dev-zero** [electronic-structure/SIRIUS#727]
Trying to build SIRIUS using Spack with sirius+cuda+scalapack+vdwxc+elpa, fails here:

```
...
5 errors found in build log:
     257    [ 81%] Building CUDA object src/CMakeFiles/sirius.dir/gpu/mul_by_veff.cu.o
     258    cd /tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-build-2yxhib3/src && /opt/nvidia/hpc_sdk/Linux_x86_64/21.9/cuda/11.4/bin/nvcc -forward-unknown-to-host-compiler -ccbin=/opt/cray/pe/craype/default/bin/CC -DH5_BUILT_AS_DYNAMIC_LIB -DSIRIUS_CUDA -DSIRIUS_ELPA -DSIRIUS_GPU -DSIRIUS_G
            PU_MEMORY_ALIGMENT=512 -DSIRIUS_HAVE_VDWXC_MPI -DSIRIUS_PROFILE -DSIRIUS_SCALAPACK -DSIRIUS_USE_MEMORY_POOL -DSIRIUS_USE_VDWXC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200809L -Dsirius_EXPORTS -I/tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyz
            x5qhnz3ntz/spack-src/src -I/tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-src/src/SDDK -I/tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-build-2yxhib3/src -I/tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3nt
            z/spack-build-2yxhib3/src/mod_files -isystem=/global/u2/m/muellert/spack/opt/spack/cray-sles15-zen3/gcc-10.3.0/spfft-1.0.6-hiab4calnyotodtnpdxbgi6tfcwyemex/include -isystem=/global/u2/m/muellert/spack/opt/spack/cray-sles15-zen3/gcc-10.3.0/spla-1.5.3-fhwe75ugzwuta4oac7mdboo24ofsgwqh/include -isystem=/global/u2/m/muellert/sp
            ack/opt/spack/cray-sles15-zen3/gcc-10.3.0/libxc-5.1.7-u5iskd5v6fy7yb6wadqsvkm4uw4o3mnc/include -isystem=/global/u2/m/muellert/spack/opt/spack/cray-sles15-zen3/gcc-10.3.0/spglib-1.16.1-4fphwbsezw4plwt2teejk43m7w7kanmu/include -isystem=/global/u2/m/muellert/spack/opt/spack/cray-sles15-zen3/gcc-10.3.0/hdf5-1.12.1-3trct2zkwad4
            maodcwunr6nns2sb62uu/include -isystem=/global/u2/m/muellert/spack/opt/spack/cray-sles15-zen3/gcc-10.3.0/elpa-2021.11.001-rejq3kafoxrohfixcczx5jfaimgomskc/include/elpa_openmp-2021.11.001/elpa -isystem=/global/u2/m/muellert/spack/opt/spack/cray-sles15-zen3/gcc-10.3.0/libvdwxc-0.4.0-rjpote4lfunx6ubsz4vqi3po3lmae3ix/include -i
            system=/opt/nvidia/hpc_sdk/Linux_x86_64/21.9/cuda/11.4/include -O3 -DNDEBUG -Xcompiler=-fPIC -std=c++14 -MD -MT src/CMakeFiles/sirius.dir/gpu/mul_by_veff.cu.o -MF CMakeFiles/sirius.dir/gpu/mul_by_veff.cu.o.d -x cu -c /tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-src/src/gpu/mul_b
            y_veff.cu -o CMakeFiles/sirius.dir/gpu/mul_by_veff.cu.o
     259    [ 82%] Building CUDA object src/CMakeFiles/sirius.dir/gpu/mul_veff_with_phase_factors.cu.o
     260    cd /tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-build-2yxhib3/src && /opt/nvidia/hpc_sdk/Linux_x86_64/21.9/cuda/11.4/bin/nvcc -forward-unknown-to-host-compiler -ccbin=/opt/cray/pe/craype/default/bin/CC -DH5_BUILT_AS_DYNAMIC_LIB -DSIRIUS_CUDA -DSIRIUS_ELPA -DSIRIUS_GPU -DSIRIUS_G
            PU_MEMORY_ALIGMENT=512 -DSIRIUS_HAVE_VDWXC_MPI -DSIRIUS_PROFILE -DSIRIUS_SCALAPACK -DSIRIUS_USE_MEMORY_POOL -DSIRIUS_USE_VDWXC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200809L -Dsirius_EXPORTS -I/tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyz
            x5qhnz3ntz/spack-src/src -I/tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-src/src/SDDK -I/tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-build-2yxhib3/src -I/tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3nt
            z/spack-build-2yxhib3/src/mod_files -isystem=/global/u2/m/muellert/spack/opt/spack/cray-sles15-zen3/gcc-10.3.0/spfft-1.0.6-hiab4calnyotodtnpdxbgi6tfcwyemex/include -isystem=/global/u2/m/muellert/spack/opt/spack/cray-sles15-zen3/gcc-10.3.0/spla-1.5.3-fhwe75ugzwuta4oac7mdboo24ofsgwqh/include -isystem=/global/u2/m/muellert/sp
            ack/opt/spack/cray-sles15-zen3/gcc-10.3.0/libxc-5.1.7-u5iskd5v6fy7yb6wadqsvkm4uw4o3mnc/include -isystem=/global/u2/m/muellert/spack/opt/spack/cray-sles15-zen3/gcc-10.3.0/spglib-1.16.1-4fphwbsezw4plwt2teejk43m7w7kanmu/include -isystem=/global/u2/m/muellert/spack/opt/spack/cray-sles15-zen3/gcc-10.3.0/hdf5-1.12.1-3trct2zkwad4
            maodcwunr6nns2sb62uu/include -isystem=/global/u2/m/muellert/spack/opt/spack/cray-sles15-zen3/gcc-10.3.0/elpa-2021.11.001-rejq3kafoxrohfixcczx5jfaimgomskc/include/elpa_openmp-2021.11.001/elpa -isystem=/global/u2/m/muellert/spack/opt/spack/cray-sles15-zen3/gcc-10.3.0/libvdwxc-0.4.0-rjpote4lfunx6ubsz4vqi3po3lmae3ix/include -i
            system=/opt/nvidia/hpc_sdk/Linux_x86_64/21.9/cuda/11.4/include -O3 -DNDEBUG -Xcompiler=-fPIC -std=c++14 -MD -MT src/CMakeFiles/sirius.dir/gpu/mul_veff_with_phase_factors.cu.o -MF CMakeFiles/sirius.dir/gpu/mul_veff_with_phase_factors.cu.o.d -x cu -c /tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5
            qhnz3ntz/spack-src/src/gpu/mul_veff_with_phase_factors.cu -o CMakeFiles/sirius.dir/gpu/mul_veff_with_phase_factors.cu.o
     261    In file included from /tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-src/src/linalg/eigensolver.cpp:26:
     262    /tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-src/src/linalg/eigenproblem.hpp: In member function 'virtual int Eigensolver_elpa::solve(ftn_int, ftn_int, sddk::dmatrix<double>&, double*, sddk::dmatrix<double>&)':
  >> 263    /tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-src/src/linalg/eigenproblem.hpp:677:9: error: 'elpa_eigenvectors_d' was not declared in this scope; did you mean 'elpa_invert_trm_d'?
     264      677 |         elpa_eigenvectors_d(handle, A__.at(memory_t::host), w.get(), Z__.at(memory_t::host), &error);
     265          |         ^~~~~~~~~~~~~~~~~~~
     266          |         elpa_invert_trm_d
     267    /tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-src/src/linalg/eigenproblem.hpp: In member function 'virtual int Eigensolver_elpa::solve(ftn_int, ftn_int, sddk::dmatrix<std::complex<double> >&, double*, sddk::dmatrix<std::complex<double> >&)':
  >> 268    /tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-src/src/linalg/eigenproblem.hpp:742:9: error: 'elpa_eigenvectors_dc' was not declared in this scope; did you mean 'elpa_invert_trm_dc'?
     269      742 |         elpa_eigenvectors_dc(handle, (CT*)A__.at(memory_t::host), w.get(), (CT*)Z__.at(memory_t::host), &error);
     270          |         ^~~~~~~~~~~~~~~~~~~~
     271          |         elpa_invert_trm_dc
  >> 272    make[2]: *** [src/CMakeFiles/sirius.dir/build.make:919: src/CMakeFiles/sirius.dir/linalg/eigensolver.cpp.o] Error 1
     273    make[2]: *** Waiting for unfinished jobs....
     274    make[2]: Leaving directory '/tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-build-2yxhib3'
  >> 275    make[1]: *** [CMakeFiles/Makefile2:227: src/CMakeFiles/sirius.dir/all] Error 2
     276    make[1]: Leaving directory '/tmp/muellert/spack-stage/spack-stage-sirius-7.3.0-2yxhib35wjzulupx3f6gyzx5qhnz3ntz/spack-build-2yxhib3'
  >> 277    make: *** [Makefile:139: all] Error 2
```

**Comment by toxa81** [comment]
This should be fixed in 7.3.1 https://github.com/electronic-structure/SIRIUS/releases

**Comment by dev-zero** [comment]
@toxa81 thanks, will try. Although, we've released the CP2K v9.1 toolchain with ELPA 2021.11.00 and SIRIUS 7.3.0 and there it seemed to have worked 🤔

**Comment by dev-zero** [comment]
Ok, verified, thanks! Still bit puzzled why it worked in the past...
